examples/basic-tcp.yaml

```diff
@@ -0,0 +1,44 @@
+kind: GatewayClass
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: acme-lb
+spec:
+  controller: acme.io/gateway-controller
+  parametersRef:
+    name: acme-lb-configmap
+    group: core
+    resource: configmaps
+---
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: my-gateway
+  namespace: default
+spec:
+  class: acme-lb
+  listeners:  # Use GatewayClass defaults for listener definition.
+  - protocol: TCP
+    port: 8080
+    routes:
+      routeNamespaces:
+        namespaceSelector: {}
+        onlySameNamespace: false
+      routeSelector:
+        matchLabels:
+          "app": "foo"
+---
+kind: TCPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: tcp-app-1
+  namespace: default
+  labels:
+    app: foo
+spec:
+  rules:
+  - action:
+      forwardTo:
+        targetRef:
+          name: my-service
+          group: core
+          resource: services
```

Comment:
Rules is an array but we could get away with it being a single rule as well (or skip the rule-level entirely for TCPRoute). It will be useful if we add SNI back in and reuse TCPRoute for TLS routing (with and without TLS termination at G/W level). 
We should keep this as is for now and come to it later.

Reply:
true -- harry -- does it makes sense to merge this w/ expectation for follow up?